include_physics/ideal_MHD.h

```diff
@@ -387,14 +387,300 @@ inline void init_fields(uint ix, uint iy, uint iz, global REAL* u) {
 // Surface terms
 //--------------------------------------------------------------------------------------------------
 
+#ifndef USE_PERIODIC
+
+#define sq(x) ((x)*(x))
+#define PLUS(x) (max(x, 0.0))
+
+inline void calc_flux_f(REAL *u, REAL *flux) {
+  REAL BB = u[Field_Bx]*u[Field_Bx] + u[Field_By]*u[Field_By] + u[Field_Bz]*u[Field_Bz];
+  flux[Field_rho] = u[Field_rho_ux];
+  flux[Field_rho_ux] = u[Field_rho_ux]*u[Field_ux] + u[Field_p] + 0.5 * BB - u[Field_Bx]*u[Field_Bx];
+  flux[Field_rho_uy] = u[Field_rho_ux] * u[Field_uy] - u[Field_Bx]*u[Field_By];
+  flux[Field_rho_uz] = u[Field_rho_ux] * u[Field_uz] - u[Field_Bx]*u[Field_Bz];
+  flux[Field_E] = u[Field_ux]*(u[Field_E] + u[Field_p] + 0.5*BB)-u[Field_Bx]*(u[Field_Bx]*u[Field_ux] + u[Field_By]*u[Field_uy] + u[Field_Bz]*u[Field_uz]);
+//flux[Field_Bx] = 0;
```

Comment:
Here, you assume explicitly that `flux` is initialised as zero. It might be good to write overwrite possible non-zero values explicitly, e.g. if some local storage is reused for multiple dimensions.

Reply:
Thanks for the comment.